apps/costcontrol/js/settings/limitdialog.js

```diff
@@ -33,8 +34,10 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
       var dataLimit = Common.getDataLimit({'dataLimitValue': value,
                                            'dataLimitUnit': currentUnit});
       SimManager.requestDataSimIcc(function(dataSim) {
-        addNetworkUsageAlarm(Common.getDataSIMInterface(dataSim.iccId),
-                             dataLimit);
+        var simInterface = Common.getDataSIMInterface(dataSim.iccId);
+        var self = this;
+        addNetworkUsageAlarm(simInterface, dataLimit, null,
+                             checkDataLimit.call(self, dataLimit));
```

Comment:
This is not a function but a call. if you would want to provide a bound-to-self version your should use `.bind(self, dataLimit)`. Notice `this` is undefined in the context the function is passed. What you need is to bind the function in `requestDataSimIcc()` call to bind, then simply use bind with `this` or an arrow function.